File: examples/libf3d/python/render-image/check.py
import sys
import f3d

if __name__ == "__main__":
    if len(sys.argv) != 3:
        sys.exit(1)

    img_0 = f3d.Image(sys.argv[1])
    img_1 = f3d.Image(sys.argv[2])

    diff = f3d.Image()
    error = 0.0

    result = img_0.compare(img_1, 50, diff, error)

    sys.exit(0 if result else 1)

File: examples/libf3d/cpp/render-image/check.cxx
#include <f3d/image.h>

int main(int argc, char** argv)
{
  // Check inputs
  if (argc != 3)
  {
    return EXIT_FAILURE;
  }

  // Open images
  f3d::image img0(argv[1]);
  f3d::image img1(argv[2]);

  // Compare them
  double error;
  return img0.compare(img1, 50, error) ? EXIT_SUCCESS : EXIT_FAILURE;
}
